File: movies-recommender-service/src/main/kotlin/org/miejski/movies/recommender/infrastructure/dbstate/assertions/PersonIndexAssertion.kt
package org.miejski.movies.recommender.infrastructure.dbstate.assertions

import org.miejski.movies.recommender.infrastructure.dbstate.Neo4jStateAssertion

class PersonIndexAssertion : Neo4jStateAssertion {

    override fun isOK(): Boolean {
        return false
    }

    override fun queryToExecute(): String {
        return "start_state/person_index.cypher"
    }

    override fun name(): String {
        return "PersonIndexAssertion"
    }
}

File: deployment/scripts/neo4j_state/assertions/person_index_assertion.py
from neo4j_state.assertions.simple_cypher_state_assertion import SimpleCypherStateAssertion


class PersonIndexAssertion(SimpleCypherStateAssertion):

    def play(self, Neo4jCypherExecutor):
        self.execute_query(Neo4jCypherExecutor, self.query_to_execute(), self.arguments())

    def is_ok(self, Neo4jCypherExecutor):
        return False

    def query_to_execute(self):
        return "person_index.cypher"

    def arguments(self):
        return {}
